Frameworks/Foundation/NSScanner.mm

```diff
@@ -25,547 +25,295 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-#include "Starboard.h"
-#include "StubReturn.h"
-#include "Foundation/NSString.h"
-#include "Foundation/NSCharacterSet.h"
-#include "Foundation/NSScanner.h"
-#include <limits.h>
+#import <Foundation/NSScanner.h>
+#import <Foundation/NSCharacterSet.h>
+#import <Foundation/NSDecimalNumber.h>
+#import <Foundation/NSString.h>
+
+#import <Starboard.h>
+
+#import <cctype>
+#import <functional>
+#import <limits>
+#import <vector>
+
+// Helper function for implementing a scan function for a numeric type using a CRT function such as wcstoll
+template <typename TNumeric>
+static BOOL __scanNumeric(NSScanner* scanner,
```

Comment:
nit: make templated on lambda so we don't have to use std::function, and then just inline the lambdas into the __scanNumeric calls